fmriprep/workflows/bold/base.py

```diff
@@ -438,20 +432,32 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
         use_fieldwarp=(fmaps is not None or use_syn),
         name='bold_bold_trans_wf'
     )
+    bold_bold_trans_wf.inputs.inputnode.name_source = ref_file
 
     # SLICE-TIME CORRECTION (or bypass) #############################################
     if run_stc is True:  # bool('TooShort') == True, so check True explicitly
         bold_stc_wf = init_bold_stc_wf(name='bold_stc_wf', metadata=metadata)
         workflow.connect([
-            (bold_reference_wf, bold_stc_wf, [('outputnode.bold_file', 'inputnode.bold_file'),
-                                              ('outputnode.skip_vols', 'inputnode.skip_vols')]),
+            (bold_reference_wf, bold_stc_wf, [
+                ('outputnode.skip_vols', 'inputnode.skip_vols')]),
             (bold_stc_wf, boldbuffer, [('outputnode.stc_file', 'bold_file')]),
         ])
+        if not multiecho:
+            workflow.connect([
+                (bold_reference_wf, bold_stc_wf, [
+                    ('outputnode.bold_file', 'inputnode.bold_file')])])
+        else:  # for meepi, iterate through stc_wf for all workflows
+            meepi_echos = boldbuffer.clone(name='meepi_echos')
+            meepi_echos.iterables = ('bold_file', bold_file)
+            workflow.connect([
+                (meepi_echos, bold_stc_wf, [('bold_file', 'inputnode.bold_file')])])
     else:  # bypass STC from original BOLD to the splitter through boldbuffer
-        workflow.connect([
-            (bold_reference_wf, boldbuffer, [
-                ('outputnode.bold_file', 'bold_file')]),
-        ])
+        if not multiecho:
+            workflow.connect([
+                (bold_reference_wf, boldbuffer, [
+                    ('outputnode.bold_file', 'bold_file')])])
+        else:  # for meepi, iterate over all meepi echos to boldbuffer
+            boldbuffer.iterables = ('bold_file', bold_file)
```

Comment:
Sorry, I don't understand this suggestion-- I don't think I can have an `elif / else` block without an `if` statement ?

Reply:
Sorry for being unclear. I mean the current structure is:

```Python
if run_stc is True:
    # block A
else:
    if not multiecho:
        # block B
    else:
        # block C
```

This could become either:

```Python
if run_stc is True:
    # block A
elif not multiecho:
    # block B
else:
   # block C
```

or:

```Python
if run_stc is True:
    # block A
elif multiecho:
    # block C
else:
    # block B
```

Reply:
I see, but there are really four options, no ? It's all combinations of `run_stc is True` and `multiecho is True`. So I could do something like:

```
if run_stc is True and multiecho is True:
    # block A
elif run_stc is False and multiecho is True:
    # block C
...
```